dnf/logging.py

```diff
@@ -180,7 +180,7 @@ def _presetup(self):
 
         # setup stderr
         stderr = logging.StreamHandler(sys.stderr)
-        stderr.setLevel(WARNING)
+        stderr.setLevel(DEBUG)
```

Comment:
Hm, so the level is set in `Logging._presetup` and then in `Logging._setup`. But there it's much later, after the configuration is loaded. So it makes sense to set the level to `DEBUG` in `Logging._presetup`.

However, it will mean always logging the configuration parsing messages. Originally, PR https://github.com/rpm-software-management/dnf/pull/610 changed it, so that these messages were only shown in verbose mode. I am not sure how it got lost, but maybe the log level could be adjusted also after parsing CLI options, i.e. before parsing the configuration files. Therefore, showing the configuration parsing messages would at least respect the `--verbose` CLI option. But it still of course won't respect the configuration option, so it might be even more confusing.

I don't know, maybe I am just overthinking this. What's your take on this?

Reply:
IMO instead of lowering the debug level, the level of the configuration error messages should be increased to either INFO or WARNING, depending on how we evaluate the severity. The PR also changes the working from "Unknown" to "Invalid", so I'm not sure, I would pick INFO for "Unknown" and WARNING for "Invalid".

Reply:
Yes, make them WARNING is another option, but I think there were some problems with that and that was the reason for the PR #610 . INFO won't change much, because it still won't be shown.

Reply:
We can also change the level here from DEBUG to INFO, that is a reasonable default. DEBUG is not, that should only be turned on explicitly for debugging.

Reply:
Thanks for the feedback guys. Thinking about it once more, I agree that using higher loglevel for invalid config values is appropriate. The unknown variables are just skipped so I'm leaving them on DEBUG.
We might even want to make dnf stop and error out in case of invalid config option. I mean in the main config case. The repositories configs will just print an error and continue. What do you think?
As far as https://github.com/rpm-software-management/dnf/pull/610 is concerned - according to the commit message it was about unknown options. Those are not affected.

Reply:
I think the error level is appropriate, but I don't see why there should be a difference between the main config and repositories configs, so I'd leave it as it is now.

@lukash do you have any objections against merging it as it is?

Reply:
No objections, this is fine :slightly_smiling_face: